client-java/src/main/java/uk/gov/gchq/palisade/client/internal/resource/WebSocketListener.java

```diff
@@ -0,0 +1,285 @@
+            return func.apply(new Item.Builder()).build();
+        }
+
+        /**
+         * Returns the type of this message
+         *
+         * @return the {@link WebSocketMessageType}
+         */
+        WebSocketMessageType getType();
+
+        /**
+         * Returns the headers for this message an empty map if there are none
+         *
+         * @return the headers for this message an empty map if there are none
+         */
+        Map<String, String> getHeaders();
+
+        /**
+         * Returns the body of this message or empty if there is none. At this point
+         * there should only be a body for {@link WebSocketMessageType} of RESOURCE.
+         *
+         * @return the body of this message or empty if there is none
+         */
+        Optional<Object> getBody();
+
+    }
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(WebSocketListener.class);
+
+    private final ObjectMapper objectMapper;
+    private final Consumer<WebSocketMessage> handler;
+    private final String token;
+
+    /**
+     * A {@code ResourceClient} manages the passing of messages to/from a websocket
+     * server
+     *
+     * @param setup The setup for this listener
+     */
+    public WebSocketListener(final ResourceClientListenerSetup setup) {
+        Checks.checkNotNull(setup);
+        this.token = setup.getToken();
+        this.handler = setup.getEventsHandler();
+        this.objectMapper = setup.getObjectMapper();
+    }
+
+    /**
+     * Helper method to create a {@link Item} using a builder function
+     *
+     * @param func The builder function
+     * @return a newly created {@code RequestId}
+     */
+    public static WebSocketListener createResourceClientListener(
+        final UnaryOperator<ResourceClientListenerSetup.Builder> func) {
+        return new WebSocketListener(func.apply(new ResourceClientListenerSetup.Builder()).build());
+    }
+
+    @Override
+    public void onError(final WebSocket ws, final Throwable error) {
+        LOGGER.debug("A {} exception was thrown.", error.getCause().getClass().getSimpleName());
+        LOGGER.debug("Message: {}", error.getLocalizedMessage());
+    }
+
+    @Override
+    public void onOpen(final WebSocket ws) {
+        Listener.super.onOpen(ws);
+        LOGGER.debug("OPEN: WebSocket Listener has been opened for requests.");
+        send(ws, b -> b.type(WebSocketMessageType.CTS));
+    }
+
+    @Override
+    public CompletionStage<?> onText(final WebSocket ws, final CharSequence data, final boolean last) {
+
+        var text = data.toString();
+
+        ws.request(1); // omit this and no methods are called listener
+
+        Item item;
+        try {
+            item = objectMapper.readValue(text, Item.class);
+        } catch (JsonProcessingException e) {
+            onError(ws, e);
+            return null;
+        }
+
+        LOGGER.debug("RCVD: {}", item);
```

Comment:
```suggestion
        LOGGER.debug("RECV: {}", item);
```
Your code seems to fall towards a present-tense style

Reply:
This is past tense as the item has been received from the websocket at this point.